cmd/start.go

```diff
@@ -218,964 +196,924 @@ func setupStartCommand() *cli.Command {
 	} else {
-		if output, err := RunCMD(pastelServicePath, fmt.Sprintf("--config-file=%s", workDirPath)); err != nil {
-			log.WithContext(ctx).Errorf("%s start was failed! : %s", commandName, output)
+		masternodeConfPath = filepath.Join(workDirPath, "masternode.conf")
+		masternodeConfPathBackup = filepath.Join(workDirPath, "masternode_%s.conf")
+	}
+	if _, err := os.Stat(masternodeConfPath); err == nil { // if masternode.conf File exists , backup
+		oldFileName := masternodeConfPath
+		currentTime := time.Now()
+		backupFileName := fmt.Sprintf(masternodeConfPathBackup, currentTime.Format("2021-01-01-23-59-59"))
+		err := os.Rename(oldFileName, backupFileName)
+		if err != nil {
 			return err
 		}
-	}
 
-	return nil
-}
-func runPastelServiceRemote(ctx context.Context, config *configs.Config, tool constants.ToolType, client *utils.Client) (err error) {
-	commandName := filepath.Base(string(tool))
-	log.WithContext(ctx).Infof("Remote:::Starting %s", commandName)
+	}
 
-	remoteWorkDirPath, remotePastelExecPath, remoteOsType, err := getRemoteInfo(config, client)
+	confFile, err := os.Create(masternodeConfPath)
+	confFile.Write(confData)
 	if err != nil {
 		return err
 	}
+	defer confFile.Close()
 
-	switch tool {
-	case constants.RQService:
-		workDirPath := filepath.Join(string(remoteWorkDirPath), "rqservice", "rqservice.toml")
-		workDirPath = strings.ReplaceAll(workDirPath, "\\", "/")
-
-		pastelRqServicePath := filepath.Join(string(remotePastelExecPath), constants.PastelRQServiceExecName[constants.OSType(string(remoteOsType))])
-		pastelRqServicePath = strings.ReplaceAll(pastelRqServicePath, "\\", "/")
-
-		go client.Cmd(fmt.Sprintf("%s %s", pastelRqServicePath, fmt.Sprintf("--config-file=%s", workDirPath))).Run()
-
-		time.Sleep(10000 * time.Millisecond)
-
-	}
-
-	log.WithContext(ctx).Infof("Remote:::The %s started succesfully!", commandName)
 	return nil
 }
 
-func runSuperNodeRemote(ctx context.Context, config *configs.Config, client *utils.Client /*, extIP string, pastelid string*/) (err error) {
-	log.WithContext(ctx).Info("Remote:::Starting supernode")
-	log.WithContext(ctx).Debug("Remote:::Configure supernode setting")
+func updateMasternodeConfFile(confData map[string]interface{}, config *configs.Config) (result bool, err error) {
+	workDirPath := config.WorkingDir
+	var masternodeConfPath string
 
-	log.WithContext(ctx).Info("Remote:::Configuring supernode was finished")
-	log.WithContext(ctx).Info("Remote:::Start supernode")
+	if config.Network == "testnet" {
+		masternodeConfPath = filepath.Join(workDirPath, "testnet3", "masternode.conf")
+	} else {
+		masternodeConfPath = filepath.Join(workDirPath, "masternode.conf")
+	}
 
-	remoteWorkDirPath, remotePastelExecPath, remoteOsType, err := getRemoteInfo(config, client)
+	// Read ConfData from masternode.conf
+	confFile, err := ioutil.ReadFile(masternodeConfPath)
 	if err != nil {
-		return err
+		return false, err
```

Comment:
it's better to wrap error with detail off error.
`return false, errros.Errorf("failed to read masternode cofinig file: %v", err)`